File: Leetcode Solutions/Two Sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap = dict()
        for i in range(len(nums)):
            if nums[i] not in hmap:
                hmap[nums[i]] = i
        for i in range(len(nums)):
            key = target - nums[i]
            if (key) in hmap:
                if hmap[key] != i:
                    return [i, hmap[key]]

File: CodingInterview/Leetcode Solutions/Challenge/Solutions/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap{std::make_pair(nums[0], 0)};
        
        for (size_t i = 1; i < nums.size(); ++i){
            auto it = hashmap.find(target - nums[i]);
            if (it != hashmap.end() && it->second != i){
                return {it->second, i};
            }
            hashmap[nums[i]] = i;
        }
        
        return {};
    }
};
